lib/trivial.nix

```diff
@@ -31,6 +68,18 @@ rec {
   /* boolean “and” */
   and = x: y: x && y;
 
+  /* bitwise “and” */
+  bitAnd = builtins.bitAnd or zipIntBits (a: b: if a==1 && b==1 then 1 else 0);
```

Comment:
I don't get it. Are these functions available as builtins or not ? Or is that just a hopeful preparation for future builtins?

Reply:
Not yet available in ```nix-2.0.4``` (default in ```nixpkgs```), only in ```nix``` built from git sources.
And probably will never be available in ```nix1```

I see it is quite common practice, for example there is 
```
any = builtins.any or (pred: foldr (x: y: if pred x then true else y) false);
```
and few others alike

Reply:
I see. Let's hope someone remembers to clean this up when the builtins become available :smiley: